chain-signatures/node/src/storage/protocol_storage.rs

```diff
@@ -679,6 +679,78 @@ impl<A: ProtocolArtifact> ProtocolStorage<A> {
         }
     }
 
+    /// Return a taken artifact back to the available pool.
+    pub async fn recycle_mine(&self, me: Participant, taken: ArtifactTaken<A>) -> bool {
```

Comment:
Adding this code breaks an important architectural invariant we had so far: Triples and Presignatures only leave storage once

I understand this invariant was introduced before I joined to ensure that Ts and Ps are only used once.
Your change doesn't break that property, which is ultimately what counts.

But if we break the current invariant, what can we replace it with? Making sure the current change doesn't break the thing we are thinking about right now is easy. Making sure no future changes accidentally break things when we are not thinking about it, is hard. That's why being strict about invariants can be really useful.

Personally, I'd prefer a solution where we don't need this recycling mechanism.

Reply:
We will need to get back to this, I've created this issue not to forget about this change: https://github.com/sig-net/mpc/issues/611